core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java

```diff
@@ -426,6 +426,12 @@ public void append(Key key, Value value) throws IOException {
         throw new IllegalArgumentException("Keys appended out-of-order.  New key " + key + ", previous key " + prevKey);
       }
 
+      // 5 bytes of overhead for fields: value, row, colFamily, colQualifier, colVisibility, and 8 for the long timestamp
+      if (((long) key.getSize() + (long) value.getSize() + (5L * 5L + 8L)) >= Integer.MAX_VALUE) {
```

Comment:
Could make `(5L * 5L + 8L)` a constant in RFile and then Key could also use that constant.  The constant could have a comment about its purpose then the code that uses it does not need that constant.  I am thinking it would be better to put the constant in RFile instead of Key because Key is in the public API, so the constant would end up in the API if in Key.